doc/api/esm.md

```diff
@@ -95,16 +95,42 @@ When loaded via `import` these modules will provide a single `default` export
 representing the value of `module.exports` at the time they finished evaluating.
 
 ```js
-import fs from 'fs';
-fs.readFile('./foo.txt', (err, body) => {
+// foo.js
+module.exports = { one: 1 };
+
+// bar.js
+import foo from './foo.js';
+foo.one === 1; // true
+```
+
+Builtin modules will provide named exports of their public API, as well as a
+default export which can be used for, among other things, modifying the named
+exports.
```

Comment:
we should make a note about getter/setters being defined in place of named exports on builtins as potentially leading to unexpected behavior since the sync is only done on get/set operations on the builtin module:

```mjs
import fs, {readFile} from 'fs';
import assert from 'assert';
let i = 10;
Object.defineProperty(fs, 'readFile', {
  get() { return i; }
});
i++;
const a = readFile;
const b = fs.readFile;
const c = readFile;
assert.equal(a, b);
assert.equal(b, c);
```

I'd be reluctant to document the actual behavior here, just want to document that this is considered and edge case of behavior.

Reply:
Yep that's a good idea. Maybe just documenting when syncs are done (define/delete/set) and that anything done outside of those won't be captured. To me the above example makes sense, but I have the context of knowing the traps used to simulate live bindings.

Reply:
we could trap on `get` as well, but we don't for now. I would find it more intuitive if we trapped on `get` but have no objections to how it behaves.

Reply:
> but have no objections to how it behaves.

Cool! Keep in mind trapping on `get` wouldn't help the first assert since accessing `readFile` doesn't trigger a get trap of `fs.readFile` so `a` would still be `10` and `b` would be `11`.

Reply:
yup, we cannot have perfect sync, just interesting that `fs.readFile = fs.readFile` updates but `fs.readFile` doesn't.